### DISCUSSION TITLE
Chart custom tooltip

### DISCUSSION THREAD

**Discussion by Yonik94** [cloudscape-design/components#124] [Q&A] [answered]
How can I customize charts tooltip? (I didn't found nothing in the API, the highlight series prop not help for that).

might I can modify the data of series for tooltip or use custom component for tooltip ?

Thanks.

**Comment by rubencarvalho** [comment, marked as answer]
Hello Yonik,

thank you for reaching out.

Can you please tell us a little bit more about your use case? What type of content would you like to present in the popover/tooltip?

We currently offer a `detailPopoverContent` [API](https://cloudscape.design/components/pie-chart/?tabId=api) for the pie and donut charts which you can use to determine the details that are displayed in the popover when hovering over a segment.
Other than that, there is no support for customising the content of the charts tooltips.

I will keep this discussion open to collect any additional feedback.

Thank you!
RC

**Reply by Yonik94** [reply]
Hi Ruben,
Thanks for your answer!
Unfortunately I need the ability to customize the Line Chart component tooltip.
I wonder why the prop is only supported on specific charts, could you estimate when will it be available with Line Chart as well?

**Reply by rubencarvalho** [reply]
We currently don't have this change planned, but we can see this as being a reasonable request.
Could you please share what type of content you would like to display on the tooltip, so we can better understand the use case?

**Reply by Yonik94** [reply]
It would be very helpful if I could pass the same prop `detailPopoverContent` in Line Charts and represent additional data whenever I hover over a single line.
Worth noting, hovering the background of the chart represent the data of multiple lines, thus not leaving much room for another data, in which case the data received from the `detailPopoverContent` prop may be ignored (if there is no optimal way to show the additional data responsively).‏

**Reply by rubencarvalho** [reply]
Thank you for your reply.

We usually only take contributions for bug fixes, but because we see this change as requiring small API reviews and little to no design input, if you’re willing to open a pull request, we’ll gladly accept [your contribution](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md).
